src/metabase/driver/sql/query_processor.clj

```diff
@@ -630,18 +696,36 @@
 ;; also, we want to gracefully handle situations where the column is ZERO and just swap it out with NULL instead, so
 ;; we don't get divide by zero errors. SQL DBs always return NULL when dividing by NULL (AFAIK)
 
+(defn- safe-denominator
```

Comment:
Not strictly related to the `:mysql` change, but as I was adding new tests I noticed that we were generating insane nonsense like

```sql
x / (CASE WHERE 7.0 = 0 THEN NULL ELSE 7.0)
```

instead of 

```sql
x / 7.0
```

so I reworked this stuff. The reason we were doing it was to prevent divide by zero errors. The fix works by checking whether we're dividing by a numeric literal and emitting the result directly instead of that big spooky `CASE` statement.